src/test/scala/firrtlTests/AsyncResetSpec.scala

```diff
@@ -319,6 +319,29 @@ class AsyncResetSpec extends FirrtlFlatSpec {
     )
   }
 
+  "AsyncReset registers" should "emit 'else' case for reset even for trivial valued registers" in {
+    val withDontTouch = s"""
+      |circuit m :
+      |  module m :
+      |    input clock : Clock
+      |    input reset : AsyncReset
+      |    input x : UInt<8>
+      |    reg r : UInt<8>, clock with : (reset => (reset, UInt(123)))
+      |""".stripMargin
+    val annos = Seq(dontTouch("m.r")) // dontTouch prevents ConstantPropagation from fixing this problem
+    val result = (new VerilogCompiler).compileAndEmit(CircuitState(parse(withDontTouch), ChirrtlForm, annos))
+    result should containLines (
+       "always @(posedge clock or posedge reset) begin",
+       "if (reset) begin",
+       "r <= 8'h7b;",
+       "end else begin",
+       "r <= r;",
```

Comment:
I think what we actually want is the same constant assignment, I would guess that assigning the register to itself here will also infer a latch but I'm not certain

Reply:
I can do that. It's a bit const-proppy but if you think that's better I'm down :)

Reply:
If they can figure out that it's a self assign and treat it specially, why couldn't they figure out that the original code wasn't a latch 🤣

Reply:
Really I think we just need to find the thing that makes the linters happy - synthesis tools should constprop any of these (including the old formulation)

And I do know that assigning the constant in both places does make at least one linter happy.

Reply:
Does the self-assign also make it happy? The two versions are distinguishable in a waveform, so I'd like to do whatever obeys the principle of least surprise, though I imagine seeing a constant value from a constant register won't surprise people too much.

I will happily do either as long as nobody objects to putting a special form of const-prop in the emitter :)

Reply:
I thought the self assignment fixed this as the issue is that there's no "action" if the `posedge clock` path trips. I'd have to actually check though.

Reply:
Now it's constant on both sides.